### PR TITLE
Version Combobox in PM UI displaying wrong or empty version

### LINKED ISSUE #11734: [Bug]: PM UI version list only shows a single latest version
### NuGet Product Used

Visual Studio Package Management UI

### Product Version

17.2.0 Preview 3.0 [32407.349.d17.2] 

### Worked before?

Yes, works in 17.1.1 and 17.3.0 preview 1.0

### Impact

It's more difficult to complete my work

### Repro Steps & Context

1. Open PM UI for a .net project
1. Switch to Browse tab
1. Ensure NuGet.org is the selected source
1. Search for `azure.resourcemanager`
1. Observe the version list menu

image

It should show many versions, not just one.

I was unable to use "Report an issue" in VS because that option made VS lock up. I'm not sure how to report that.

### Verbose Logs

_No response_

### LINKED ISSUE #11802: [Bug]: The Latest Stable version does not show in the "Version" dropdown box after installing a package with not the latest version in “Updates” tab
### NuGet Product Used

Visual Studio Package Management UI

### Product Version

Dev\6.3.0.51

### Worked before?

Dev\6.3.0.15

### Impact

It bothers me. A fix would be nice

## Repro steps:  
1. Create a C# Console App (.NET core) project via VS.
2. Open PM UI of the project by right clicking the project->Manage NuGet Packages… in Solution Explorer.
3. Install a package (e.g. Newtonsoft.json) into the project with not the latest version in “Browse” tab.
4. Switch to the “Update” tab and observe the “version” dropdown box.
## Expected:
The Latest Stable version should show in the "Version" dropdown box in “Updates” tab as below screenshot.
image
## Actual: 
The installed version shows in the "Version" dropdown box in “Updates” tab as below screenshot. 
image
## Note:
1. This issue starts to repro on Dev\6.3.0.17.

### Verbose Logs

_No response_

### LINKED ISSUE #11806: [Bug]: [Bug Bash] The “Version” dropdown box is blank in “Consolidate” tab of solution-level PM UI
### NuGet Product Used

Visual Studio Package Management UI

### Product Version

Dev\6.3.0.51

### Worked before?

_No response_

### Impact

It's more difficult to complete my work

### Details about Problem
VS Version: Main\32509.107
NuGet Version: Dev\6.3.0.51
OS Version: Windows-11-Enterprise-21H2

### Repro steps:
1.Create a solution with two C# Console App (.NET Framework 4.8) projects in VS.
2.Right-Click the solution in the Solution Explorer and select the “Mange NuGet Packages for Solution”.
3.Go to “Browse” tab, select a package (e.g. Newtonsoft.Json) and install different versions (with PackageReference format) into the different projects. (e.g. install 12.0.1 into the first project, install 12.0.2 into the second project)
4.Click the “Consolidate” tab, click the package “Newtonsoft.Json” in the list and observe the “Version” dropdown box.

### Expected:
The “Version” dropdown box is showing the latest installed version (12.0.2 in this instance), the version in the drowdown list can be reselected and the "Install" button is available.

### Actual:
The “Version” dropdown box is blank, the version in the drowdown list cannot be reselected and the "Install" button is not available as below.
image

### Notes:
1. This issue starts to repro on Dev\6.3.0.10, does not repro on Dev\6.3.0.9.
2. It does not repro on .NET SDK style solution.
3. It does not repro when the package is installed with “packages.config” format in step3.

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11734
Fixes: https://github.com/NuGet/Home/issues/11806
Fixes: https://github.com/NuGet/Home/issues/11802

Regression? Last working version:

## Description

This pr fixes bugs that were related, the bug was when the SelectVersion function was called, also I removed unnecessary SelectedVersion/SelectedIndex that triggered extra calculations leading to errors

Showing only 1 Version Combobox fixed (https://github.com/NuGet/Home/issues/11734):
image

Emprt Version Combobox fixed (https://github.com/NuGet/Home/issues/11806)
image

Display correct Version in Updates Tab (https://github.com/NuGet/Home/issues/11802)
image

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by jeffkl — DISMISSED** [review]
Add a unit test for `SelectVersionPackageReferenceProject` if possible

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs [hunk 1]
```diff
@@ -684,36 +684,54 @@ protected void SelectVersion(NuGetVersion latestVersion = null)
...
-                    FirstDisplayedVersion = SelectedVersion;
+                    SelectVersionPackageReferenceProject(latestVersion);
                 }
                 else
                 {
-                    var possibleVersions = _versions.Where(v => v != null);
-                    SelectedVersion =
-                        possibleVersions.FirstOrDefault(v => v.Version.Equals(_searchResultPackage.InstalledVersion))
-                        ?? possibleVersions.FirstOrDefault(v => v.IsValidVersion);
+                    SelectVersionNonPakageReferenceProject(latestVersion);
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: Typo in the method name: 

`SelectVersionNonPakageReferenceProject` to `SelectVersionNonPackageReferenceProject`

**Review by nkolev92 — DISMISSED** [review]

**Comment by zivkan** [comment]
@martinrrm the PR title makes me think this is fixing bugs in restore. Can you please update the title, so that once it's merged into dev, the commit history is more clear about what feature area the commit is relevant to?

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs [hunk 2]
```diff
@@ -362,12 +362,175 @@ public async void IsSelectedVersionCorrect_WhenPackageStyleIsPackageReference_An
             Assert.Equal(model.Versions.FirstOrDefault(), displayVersion);
         }
 
+        [Theory]
+        [InlineData("*", "2.10.0")]
+        [InlineData("1.*", "2.10.0")]
+        [InlineData("(0.1,3.4)", "2.10.0")]
+        [InlineData("2.10.0", "2.10.0")]
+        public async void IsSelectedVersionCorrect_WhenPackageStyleIsPackageReference_And_CustomVersion_UpdatesTab(string allowedVersions, string installedVersion)
```
Comment (nkolev92) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs]: naming: The name should probably be flipped based on https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#unit-test-method-naming

**Review by nkolev92 — DISMISSED** [review]
Please correct the title as @zivkan suggested.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs (hunk 2 above)
Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs]: Same applies to some of the other tests.

Comment (martinrrm) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs]: I updated the test name, you think this is better? Not sure if I should leave the `When`

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**